src/librustc/ty/mod.rs

```diff
@@ -2693,13 +2677,29 @@ fn def_span<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId) -> Span {
     tcx.hir.span_if_local(def_id).unwrap()
 }
 
+
+
+/// If the given def ID describes an item belonging to a trait,
+/// return the ID of the trait that the trait item belongs to.
+/// Otherwise, return `None`.
+fn trait_of_item<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId) -> Option<DefId> {
```

Comment:
I believe your error is coming from here. I actually implemented this also... It has to do with you not handling the case where it's not part of local crate. See https://github.com/rust-lang/rust/pull/41724/files#diff-c888afeccd78f53e5ea3eb04089e46ddR2683